instagram/src/components/PostPage/PostPage.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react'
+import dummyData from '../../dummy-data';
+import PostContainer from '../PostContainer/PostContainer';
+import SearchBar from '../SearchBar/SearchBar';
+
+export default class PostPage extends Component {
+
+    constructor(props){
+        super(props);
+        this.state = {
+            dummy: [],
+            comments: [],
+            filterPosts: []
+        }
+    }
+    
+    componentDidMount() {
+            this.setState({
+                dummy: dummyData,
+                comments: dummyData.map(data => data.comments)
+            })
+        }
+
+    SearchPostHandler = e => {
+        e.preventDefault()
+        const posts = this.state.dummy.filter(post => {
+            if (post.username.includes(e.target.value)){
+                return post;
```

Comment:
Can be refactored. All you need to return here is the condition because `.filter()` will return the current item if that condition is met. 
```suggestion
                const posts = this.state.dummy.filter(post => post.username.includes(e.target.value))
```